lutris/game_actions.py

```diff
@@ -29,20 +29,26 @@
 
 
 def get_game_actions(game, window, application=None):
+    games = [game]
     if game.is_db_stored:
-        return GameActions(game, window, application)
+        return GameActions(games, window, application)
 
     if game.service:
-        return ServiceGameActions(game, window, application)
+        return ServiceGameActions(games, window, application)
 
-    return BaseGameActions(game, window, application)
+    return BaseGameActions(games, window, application)
+
+def get_multiple_game_actions(games, window):
+    """Game actions for multiple game selections"""
+    return MultiGameActions(games, window)
 
 
 class BaseGameActions:
-    def __init__(self, game, window, application=None):
+    def __init__(self, games, window, application=None):
         self.application = application or Gio.Application.get_default()
         self.window = window  # also used as a LaunchUIDelegate
-        self.game = game
+        self.game = games[0]
+        self.games = games
```

Comment:
Let's not have both self.game and self.games at the same time like this, in the base class.

Instead, set self.game in the sub-classes that support one game only (GameActions, ServiceGameActions)- they can check that there's only one. If there is exactly one, having self.game and self.games at the same time is safe; it won't matter which the code uses.

As is, you'll have code that has been overlooked (like on_game_view below) just targeting a single game, whichever was first in the selection.

Reply:
Ok. Will need to modify all the places where it uses self.game then.

Reply:
That's the idea. Find all the places that should have loops!

Reply:
Ok